src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -115,15 +112,27 @@ final case class ClientApiServiceImpl(
     val operationLabel: String = s"Retrieving client $clientId"
     logger.info(operationLabel)
 
+    def isConsumerOrProducer(organizationId: UUID, client: ManagementClient)(implicit
+      ec: ExecutionContext,
+      contexts: Seq[(String, String)]
+    ): Future[Boolean] = {
+      def isProducer(): Future[Boolean] = Future
+        .traverse(client.purposes.map(_.states.eservice.eserviceId))(catalogManagementService.getEService)
+        .map(eservices => eservices.map(_.producerId).contains(organizationId))
+
+      (client.consumerId == organizationId).pure[Future].ifM(true.pure[Future], isProducer())
```

Comment:
Cats syntax should not be used with Future, its behaviour can be unpredictable, prefer something like

```scala
if (client.consumerId == organizationId) Future.successful(true) else isProducer()
```